pkg/interface/src/views/components/Comments.tsx

```diff
@@ -40,22 +48,75 @@ export function Comments(props: CommentsProps) {
     }
   };
 
+  const onEdit = async (
+    { comment },
+    actions: FormikHelpers<{ comment: string }>
+  ) => {
+    try {
+      const commentNode = comments.children.get(bigInt(props.editCommentId));
+      const [idx, _] = getLatestCommentRevision(commentNode);
+
+      const content = scanForMentions(comment);
+      const post = createPost(
+        content,
+        commentNode.post.index,
+        parseInt(idx + 1, 10)
+      );
+      await api.graph.addPost(ship, name, post);
+      history.push(baseUrl);
+    } catch (e) {
+      console.error(e);
+      actions.setStatus({ error: e.message });
+    }
+  };
+
+  let commentContent = null;
+  if (props.editCommentId) {
+    const commentNode = comments.children.get(bigInt(props.editCommentId));
+    const [_, post] = getLatestCommentRevision(commentNode);
+    commentContent = post.contents.reduce((val, curr) => {
+      if ('text' in curr) {
+        val = val + curr.text;
+      } else if ('mention' in curr) {
+        val = val + curr.mention;
+      } else if ('url' in curr) {
+        val = val + curr.url;
+      } else if ('code' in curr) {
+        val = val + curr.code.expression;
+      }
+      return val;
+    }, '');
+  }
+
   return (
     <Col>
-      <CommentInput onSubmit={onSubmit} />
-      {Array.from(comments.children).reverse().map(([idx, comment]) => (
-        <CommentItem
-          comment={comment}
-          key={idx.toString()}
-          contacts={props.contacts}
-          api={api}
-          name={name}
-          ship={ship}
-          hideNicknames={props.hideNicknames}
-          hideAvatars={props.hideAvatars}
-          remoteContentPolicy={props.remoteContentPolicy}
+      {( !props.editCommentId ? <CommentInput onSubmit={onSubmit} /> : null )}
+      {( !!props.editCommentId ? (
+        <CommentInput
+          onSubmit={onEdit}
+          label='Edit Comment'
+          loadingText='Editing...'
+          initial={commentContent}
         />
-      ))}
+      ) : null )}
+      {Array.from(comments.children).reverse()
+        .filter(([idx, _]) => idx.toString() !== props.editCommentId)
```

Comment:
Why do we pop it out when we're editing? It's sorta confusing in practice — could we just fade it as if it's "pending" while we're editing it?

Reply:
Sure, sounds fine.